src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/JsonDescriptor.php

```diff
@@ -147,9 +147,17 @@ protected function describeCallable(mixed $callable, array $options = []): void
         $this->writeData($this->getCallableData($callable), $options);
     }
 
-    protected function describeContainerParameter(mixed $parameter, array $options = []): void
+    protected function describeContainerParameter(mixed $parameter, ContainerBuilder $containerBuilder, array $options = []): void
     {
         $key = $options['parameter'] ?? '';
+        $parameterBag = $containerBuilder->getParameterBag();
+
+        if ($deprecation = $parameterBag instanceof ParameterBag ? ($parameterBag->allDeprecated()[$key] ?? false) : false) {
+            $parameter = [
+                'value' => $parameter,
+                'deprecated' => sprintf('Deprecated since %s %s: %s', $deprecation[0], $deprecation[1], sprintf(...\array_slice($deprecation, 2))),
```

Comment:
same everywhere actually:
```suggestion
                'deprecated' => sprintf('Since %s %s: %s', $deprecation[0], $deprecation[1], sprintf(...\array_slice($deprecation, 2))),
```